### PR TITLE
ActiveMQ MessageAge plugin

### PR DESCRIPTION
**Pull request by lganzzzo**

### CODE REVIEW

**Comment by beccatortell** [comment]
Please include information in README.md. 
- What is this plugin for? 
- What does it depend on? What are its requirements? 
- How do you install and use it? Give an example if necessary. 
- What metrics does it emit?

**Comment by mdubbyap** [comment]
Need copyright on all source files and javadoc on public methods. This should always be there.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,30 @@
 # activemq-integration
+
 Query activemq for message age metrics and publish them
+
+# What is this plugin for?
+This pluging is for reporting messages age in the ActiveMQ-queues. It reports two metrics:
+- message.age.average - average message age in each queue
```
Comment (mdubbyap) [inline README.md]: types?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,30 @@
 # activemq-integration
```
Comment (mdubbyap) [inline README.md:1]: there are 16 issues with this readme.  Run mdl --warnings on your .md files before making a PR.

**Comment by mdubbyap** [comment]
Also need a license file for open source projects. check out https://github.com/signalfx/metricproxy/blob/master/LICENSE

File: README.md [hunk 3]
```diff
@@ -1,2 +1,30 @@
...
+# What is this plugin for?
+This pluging is for reporting messages age in the ActiveMQ-queues. It reports two metrics:
+- message.age.average - average message age in each queue
+- message.age.maximum - maximum message age in each queue
+
+# Requirements
+- Linux/Unix
+- ActiveMQ
+- Maven
+
```
Comment (beccatortell) [inline README.md:20]: Java version?

File: amq-message-age/src/main/java/com/signalfx/amq/CollectMessageAge.java [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+        
+        BufferedReader br = null;
+        
+        try {
+            File file = new File("messages.txt");
+            if(file.exists()){
+                file.delete();
+            }
+            Process p = Runtime.getRuntime().exec(
+                    path + " browse --amqurl " + host + " " + queue + " | grep JMSTimestamp | grep -Eo '[0-9]{1,}' >> messages.txt");
```
Comment (mdubbyap) [inline amq-message-age/src/main/java/com/signalfx/amq/CollectMessageAge.java]: this writing to a file and then reading into a file is unacceptable. so we're just calling out to another executable within activemq? not sure java is the best language for this, but i guess we're there now (i feel like python would have been better).  At worst, read the stdout of the command and process that without using a file and replace the grepping with regexes in java as well.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,30 @@
...
+- ./run.sh
+
+# "properties" file
+
+- path - path to ActiveMQ executable
+- token - SignalFX-API-TOKEN
+- sfx_host - let it be https://ingest.signalfx.com - host to send data to
+- interval - interval how often to query on ActiveMQ messages age
+- host - activemq host
+- queue_name_prifix - prefix which will be added to the queue name in the dimension defined by queue_dimension_name
```
Comment (mdubbyap) [inline README.md]: we have the limitation in generic jmx that we can't separate the broker from the queue or topic, but should do that here.  Have a separate dimensions for broker, queue and topic.

**Comment by mdubbyap** [comment]
see no reason to checkin a blank file messages.txt even if we were going to go that direction with the parsing. see feedback on why that is bad.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,30 @@
...
+- message.age.maximum - maximum message age in each queue
+
+# Requirements
+- Linux/Unix
+- ActiveMQ
+- Maven
+
+# Installation
+
+- clone repo
```
Comment (mdubbyap) [inline README.md]: nit: put the command

File: README.md (hunk 1 above)
Comment (beccatortell) [inline README.md]: What units does this report message age in? Seconds? Milliseconds?

File: amq-message-age/src/main/java/com/signalfx/amq/CollectMessageAge.java (hunk 4 above)
Comment (lganzzzo) [inline amq-message-age/src/main/java/com/signalfx/amq/CollectMessageAge.java]: fixed

File: README.md (hunk 5 above)
Comment (lganzzzo) [inline README.md]: fixed

File: README.md [hunk 7]
```diff
@@ -1,2 +1,36 @@
...
+- Maven
+
+## Installation
+
+- clone repo (`git clone https://github.com/signalfx/activemq-integration.git`)
+- Edit `<repo>/amq-message-age/properties` file to config plugin
+- `cd <repo>/amq-message-age`
+- `./run.sh`
+
+## "properties" file
```
Comment (mdubbyap) [inline README.md:28]: why is properties in quotes?

Comment (lganzzzo) [inline README.md:28]: To show that actual name of a file is "properties"

**Pull request merged**